### ISSUE TITLE
Optuna trials_dataframe() should allow user to provide for custom columns too.

### ISSUE THREAD

**Issue by adiv5** [optuna/optuna#3311]
### Motivation

Currently trials_dataframe() allows us to view all trials' information, and the hyperparams used in the process.

However its only limited to a single score "Value" which is the metric thats optimized. 

### Description

Having a functionality that allows user to define custom metrics or "alternate values" to be shown in the trial. This means that this additional metric will be implemented by the user who wishes to use it, and would be a part of objective. 

However this doesnt entail any additional metric to optimize, just the values will be shown.

For example: 

Lets say I am optimizing train loss in a NN:
```
def objective(trial):
    params ={
    .....
    }
    train_loss,train_auc = run_train(params)
    return train_loss
```

In the above, we can see that `train_auc` is the additonal metric for each run i am interested in.

I would like to have that value reported for each trial alongside the loss 

Hope it is clear to understand. Thanks

**Comment by himkt** [comment]
Hi @adiv5, thank you for raising the issue. Doesn't user attributes help you?

```python
> python
Python 3.9.10 (main, Jan 19 2022, 01:48:10)
[Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import optuna
>>> study = optuna.create_study()
[I 2022-02-14 18:55:58,189] A new study created in memory with name: no-name-eb445c05-6117-426e-936f-034f83b282df
>>> trial = study.ask()
>>> trial.set_user_attr("train_auc", 0.5)
>>> study.tell(trial, trial.suggest_float("v", 0, 1))
>>> study.trials_dataframe()
   number     value             datetime_start          datetime_complete               duration  params_v  user_attrs_train_auc     state
0       0  0.546975 2022-02-14 18:56:08.089785 2022-02-14 18:56:50.810443 0 days 00:00:42.720658  0.546975                   0.5  COMPLETE
```

You can see `user_attrs_train_auc` in the dataframe, which is recorded by `trial.set_user_attr`.
You can store a value in the user attribute that is not used for the optimization.

**Comment by adiv5** [comment]
@himkt  Thanks for the helpful response. I think I had trouble finding the `set_user_attr` method in the docs, so completely missed it.

**Comment by himkt** [comment]
I'm happy to hear that @adiv5!
Thank you for using Optuna!
